service/migration_manager.cc

```diff
@@ -173,13 +173,16 @@ future<> migration_notifier::unregister_listener(migration_listener* listener)
     return _listeners.remove(listener);
 }
 
-void migration_manager::schedule_schema_pull(const gms::inet_address& endpoint, const gms::endpoint_state& state)
+future<> migration_manager::schedule_schema_pull(const gms::inet_address& endpoint, const gms::endpoint_state& state)
 {
+    auto guard = co_await start_group0_operation();
```

Comment:
Unfortunately this will call raft barrier. Probably safe, but not what we want. May be better to call `get_group0_upgrade_state` and check that it is `use_post_raft_procedures`.

Reply:
Addressed in https://github.com/scylladb/scylladb/commit/ccd2942567627dce2b55c556b5bc3e48de0deba1

NOTE: but using pre as in `if upgrade_state is use_pre_raft_procedures then pull schema`